src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +20,51 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-ampHours ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(previousCurrent<0)
+    {
     return false;
+    }
+    else
+    {
+       return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    double timeRemaining = (BATTERY_AMP_HOUR_CAPACITY-ampHours)/previousCurrent ;
+
+    timeRemaining = qAbs(timeRemaining)/MILISECONDS_TO_HOURS;
+
+    return QTime(0, 0, 0, 0).addMSecs(timeRemaining);
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+    double newCurrent = batteryData.current;
```

Comment:
This value isn't really the new current except at the very beginning, it can be referred to as the present current in the run.